pkg/discovery/worker/garbage_collector.go

```diff
@@ -231,3 +235,70 @@ func (g *GarbageCollector) unpauseRollouts() error {
 
 	return nil
 }
+
+func (g *GarbageCollector) cleanupQuotas() error {
+	quotaList, err := g.client.CoreV1().ResourceQuotas("").List(context.TODO(), gcListOpts)
+	if err != nil {
+		glog.Errorf("Error garbage cleaning quotas: %v", err)
+		return err
+	}
+	if quotaList == nil {
+		// ignore
+		return nil
+	}
+
+	for _, quota := range quotaList.Items {
+		err := revertQuota(g.client, &quota)
+		if err != nil {
+			// we force kubeturbo to restart
+			return err
+		}
+	}
+
+	return nil
+}
+
+func revertQuota(client *kubernetes.Clientset, quota *api.ResourceQuota) error {
+	var revertedQuota *api.ResourceQuota
+	var err error
+	if quota.Annotations != nil {
+		origSpec, exists := quota.Annotations[executor.QuotaAnnotationKey]
+		if exists {
+			revertedQuota, err = executor.DecodeQuota([]byte(origSpec))
+			if err != nil {
+				// This is very unlikely but not an error which we can recover from, for example in the next run.
+				// We need to ignore this. The annotation if there still is any would then carry the opportunity
+				// for an user to use this as a reference in case a manual correction is required ever.
+				glog.Warningf("Error reverting quota while garbage collecting resources: %s/%s: %v", quota.Namespace, quota.Name, err)
+			}
+		}
+	}
+	if revertedQuota != nil {
+		// revertedQuota being nil possibly the result of the decode error above
+		// so we go ahead and try to remove the GC label alone
+		quota.Spec.Hard = revertedQuota.Spec.Hard
+	}
+	executor.RemoveGCLabelFromQuota(quota)
+	RemoveTurboAnnotionFromQuota(quota)
+
+	_, err = client.CoreV1().ResourceQuotas(quota.Namespace).Update(context.TODO(), quota, metav1.UpdateOptions{})
+	if err != nil {
+		glog.Errorf("Error reverting quota while garbage collecting resources: %s/%s: %v", quota.Namespace, quota.Name, err)
+		return err
+	}
+
+	return nil
+}
+
+func RemoveTurboAnnotionFromQuota(quota *api.ResourceQuota) {
+	annotations := quota.GetAnnotations()
+	if annotations == nil {
+		// nothing to do
+		return
+	}
+	if _, exists := annotations[executor.QuotaAnnotationKey]; exists {
+		delete(annotations, executor.QuotaAnnotationKey)
+	}
+
+	quota.SetAnnotations(annotations)
```

Comment:
nit: same as above, move it into the `exists` block.

Reply:
Done!